### PR TITLE
Hf

### PR DESCRIPTION
**Pull request by armancohan**
Supporting Huggingface Transformers

### CODE REVIEW

File: scripts/embed_papers_hf.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        batch_size = self.batch_size
+        i = 0
+        for k, d in self.data.items():
+            if (i) % batch_size != 0 or i == 0:
+                batch_ids.append(k)
+                batch.append(d['title'] + ' ' + (d.get('abstract') or ''))
+            else:
+                input_ids = self.tokenizer(batch, padding=True, truncation=True, 
+                                           return_tensors="pt", max_length=self.max_length)
+                yield input_ids.to('cuda'), batch_ids
```
Comment (sergeyf) [inline scripts/embed_papers_hf.py:36]: I'm a little confused here as `batch` is never actually yielded?

File: scripts/embed_papers_hf.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    parser.add_argument('--output', help='path to write the output embeddings file. '
+                                        'the output format is jsonlines where each line has "paper_id" and "embedding" keys')
+    parser.add_argument('--batch-size', type=int, default=8, help='batch size for prediction')
+
+    args = parser.parse_args()
+    dataset = Dataset(data_path=args.data_path, batch_size=args.batch_size)
+    model = Model()
+    results = {}
+    batches = []
+    for batch, batch_ids in tqdm(dataset.batches(), total=len(dataset) // args.batch_size):
```
Comment (sergeyf) [inline scripts/embed_papers_hf.py:69]: Similar to above comment - the output of `dataset.batches()` is `input_ids, batch_ids`, not `batch, batch_ids`?

**Review by sergeyf — APPROVED** [review]
Looks good overall but I'm confused by the output of `Dataset.batches`

File: scripts/embed_papers_hf.py (hunk 1 above)
Comment (armancohan) [reply scripts/embed_papers_hf.py:36]: `batch` is processed into `input_ids` (token ids) and the `token_ids` is yielded.
`batch_ids` is paper ids. The variable names could be improved.

Comment (sergeyf) [reply scripts/embed_papers_hf.py:36]: Ah ok I didn't read carefully enough. Thanks.

**Pull request merged**